### ISSUE TITLE
Question: Failure to change class names and odd CSS property names

### ISSUE THREAD

**Issue by rowanwork** [styled-components/jest-styled-components#59]
I am running my tests via the addon for Storybook (Storyshots).

Some of my Snapshots started failing a few days after using jest-styled-components (have just upgraded to 4.3 and there is no change). I have noticed that it doesn't seem to be intercepting all of the styled components class names (since the hash has changed on some snapshots and is not getting converted to something like `c0`).

I have also noticed weird property names like `wc0dth` and `posc0tc0on` in the failure report. Is looks like a regex is incorrectly capturing property names and not class names:

```
    -.eBc0kCs {
    +.c0MWc0eg {
    +  wc0dth: 8px;
    +  posc0tc0on: absolute;
    +  bottom: 0;
    +  left: 0;
    +}
    +
    +.fc0NsQu {
       posc0tc0on: relatc0ve;
       left: 0;
       top: 0;
       bottom: 0;
       wc0dth: 8px;
```

Looking at the snapshot file I can also see things like:
```
<dc0v
  className="card-contac0ner"
```

Some of my snapshots do have correct class names e.g. `c0`, but I am yet to determine the differences between styled components that work and ones that don't.

I am currently trying to reproduce it outside of Storybook, but in the meantime, do you have any ideas or particular information that you require?

**Comment by MicheleBertoli** [comment]
This is pretty weird, thanks for opening the issue @rowanwork.
If you could provide a non-working example that'd be great.

**Comment by MicheleBertoli** [comment]
Any news about this issue? 
Thanks!

**Comment by rowanwork** [comment]
I have not had any luck getting creating a new basic standalone failing test or isolating the failures to just one test.

For example one failing test had something like:

```
          <use
           arc0a-hc0dden={null}
           xlc0nkHref="#my-svg"
```

But if I remove all other tests then the failing test starts to look more reasonable:
```
        <use
            aria-hidden={null}
            xlinkHref="#my-svg"
```

So something very weird is happening as I start to add more tests.

There does seem to be a common theme though where the letter `i` is getting replaced with `c0`.

**Comment by MicheleBertoli** [comment]
This is definitely a class named `i` that breaks the serializer.
It'd be fantastic to have a non-working example to debug.

**Comment by rowanwork** [comment]
I have a single test that is failing. It seems to be caused by having a class name that is just a single character:

```
const Thing = styled.div`
  width: '100px';
  position: absolute;
  bottom: 0;
  left: 0;
`;

    <div>
      <div className="i">
        <Thing>Google</Thing>
      </div>
    </div>
```

It fails with:
```
    +.c1 {
    +  c0idth: '100px';
    +  position: absolute;
    +  bottom: 0;
    +  left: 0;
    +}
    +
    +<div>
    +  <div
    +    className="c0"
    +  >
    +    <div
    +      className="c1"
    +    >
    +      Google
    +    </div>
    +  </div>
    +</div>
```

**Comment by MicheleBertoli** [comment]
Thanks for the non-working example, I've been able to repro the issue and I'm fixing it.

**Comment by MicheleBertoli** [comment]
Fixed in [4.4.1](https://github.com/styled-components/jest-styled-components/blob/master/CHANGELOG.md#441---2017-08-19).
Thank you very much!
